crates/superposition_types/src/lib.rs

```diff
@@ -82,107 +88,6 @@ impl<T> Exp<T> {
-            }
-        }
-
-        impl $wrapper<$type> {
-            pub fn try_from_db(map: Map<String, Value>) -> Result<Self, String> {
-                #[cfg(feature = "disable_db_data_validation")]
-                return Ok(Self($type(map)));
-                #[cfg(not(feature = "disable_db_data_validation"))]
-                return Self::try_from(map);
-            }
-        }
-    };
-}
-
-#[derive(
-    Deserialize, Serialize, Clone, AsRef, Deref, DerefMut, Debug, Eq, PartialEq, Into,
-)]
-pub struct Overrides(Map<String, Value>);
-
-impl Overrides {
-    fn validate_data(override_map: Map<String, Value>) -> Result<Self, String> {
-        if override_map.is_empty() {
-            log::error!("Override validation error: Override is empty");
-            return Err("Override should not be empty".to_owned());
-        }
-        Ok(Self(override_map))
-    }
-}
-
-impl IntoIterator for Overrides {
-    type Item = (String, Value);
-    type IntoIter = <Map<String, Value> as IntoIterator>::IntoIter;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.0.into_iter()
-    }
-}
-
-impl_try_from_map!(Cac, Overrides, Overrides::validate_data);
-impl_try_from_map!(Exp, Overrides, Overrides::validate_data);
-
-#[derive(Deserialize, Serialize, Clone, AsRef, Deref, Debug, Eq, PartialEq, Into)]
-pub struct Condition(Map<String, Value>);
-impl Condition {
-    fn validate_data_for_cac(condition_map: Map<String, Value>) -> Result<Self, String> {
-        if condition_map.is_empty() {
-            log::error!("Condition validation error: Context is empty");
-            return Err("Context should not be empty".to_owned());
-        }
-        jsonlogic::expression::Expression::from_json(&json!(condition_map)).map_err(
-            |msg| {
-                log::error!("Condition validation error: {}", msg);
-                msg
-            },
-        )?;
-        Ok(Self(condition_map))
-    }
-
-    fn validate_data_for_exp(condition_map: Map<String, Value>) -> Result<Self, String> {
-        let condition_val = json!(condition_map);
-        let ast = jsonlogic::expression::Expression::from_json(&condition_val).map_err(
-            |msg| {
-                log::error!("Condition validation error: {}", msg);
-                msg
-            },
-        )?;
-        let dimensions = ast.get_variable_names().map_err(|msg| {
-            log::error!("Error while parsing variable names : {}", msg);
-            msg
-        })?;
-        if dimensions.contains("variantIds") {
-            log::error!("experiment's context should not contain variantIds dimension");
-            return Err(
-                "experiment's context should not contain variantIds dimension"
-                    .to_string(),
-            );
-        }
-        Ok(Self(condition_map))
-    }
-}
-
-impl_try_from_map!(Cac, Condition, Condition::validate_data_for_cac);
-impl_try_from_map!(Exp, Condition, Condition::validate_data_for_exp);
```

Comment:
INFO: Moved to [here](https://github.com/juspay/superposition/pull/264/files#diff-61d7d423a4c1cc86aa26179f4586efff51e9d8dc8a0d414868d820073e622de6R112-R154)